### ISSUE TITLE
Travis caching?

### ISSUE THREAD

**Issue by mccalluc** [parklab/refinery-platform#1365]
```
101.92s $ pip install -r requirements.txt
70.93s  $ npm install --progress false
16.28s  $ bower install --config.interactive=false
```
If much of this time is building rather than just downloading, Travis caching could save us a couple minutes on each run. See if it works?

**Comment by hackdna** [comment]
How does this caching work? How is the cache invalidated?

**Comment by mccalluc** [comment]
https://docs.travis-ci.com/user/caching/#How-does-the-caching-work%3F

First time:
- Travis gets dependencies as usual
- at the end tars them up and move to s3

Successive times:
- Copy files from s3 to instance as part of startup
- If new libraries or new versions are requested, also get them
- At the end, if there are any differences, update s3

If the time is simply network time, Travis caching won't speed things up.

**Comment by mccalluc** [comment]
We talked in person, and we won't implement this until it's past the experimental phase, at least.

But here's what I saw on my branch:
- [setting up the cache is negligible](https://travis-ci.org/parklab/refinery-platform/builds/155189888#L122)
- [pip now takes 19s](https://travis-ci.org/parklab/refinery-platform/builds/155189888#L139), down from 102s.
- npm (71s) and bower (15s) are unchanged, because I haven't figured out how to configure caching for them: Had trouble finding clear directions for these.
